### ISSUE TITLE
Slack Team

### ISSUE THREAD

**Issue by notenociz** [dwyl/start-here#77]
A [Slack](https://slack.com/) team at something like dwyl.slack.com would be useful for the dwyl community:
- to gather
- to discuss
- to collaborate on projects and debates
- to organise/plan projects
- to keep track of the community
- to socialise
- to organise meet ups in
- to try [cool tricks](http://thenextweb.com/apps/2015/02/05/13-cool-things-might-not-know-can-slack/)
- and more...

I suggest making one and adding it into the start here here for maximum impact. Also, don't most companies use Slack by now? If not, they [should](https://blog.bufferapp.com/slack-community)!

To save you time, get one for [free](https://slack.com/pricing) [here](https://slack.com/getting-started) 😀

![slack in use](https://camo.githubusercontent.com/94daec0dc7e4f343f5f9b2c23f7e5f7c42ca1d61/687474703a2f2f636c2e6c792f577276732f696d6167652e676966)

Let's do this! 
![little boy loads gun](https://lh3.googleusercontent.com/proxy/qPDCWQL_cOKUCNN0idR_1bpGxuGKNfRWA2IyMjlLPRksRtvgYJCdWUctOsf5WYJXbjlPvPg6PT9jHuCt1zYpiJ1JrSYZsAIUsZHFF8-J0KWJ1t_ECnfJWmu5oT2wa_0Q1QUnJEAYQ6ZSWCpPJgeufQp_InF165iHlDRUO_PaCk0=w480-h258-nc)

**Comment by iteles** [comment]
@enociz Thank you for suggesting this and for all your initiative lately, so great! :ok_hand: 
We totally agree that having a place where everyone can collaborate is essential (and I'm particularly  enthusiastic about the fact that you're suggesting so many different ways!). 

We actually already have [a gitter channel for general dwyl chatl](gitter.im/dwyl/chat)!    
Gitter is by no means a perfect tool but most of the people we've spoken to in our community already use it as a daily tool and there's nothing worse than having to check _multiple_ forms of communication to stay up to date - we felt that was quite a big barrier to contributions. They're also super supportive of open source initiatives like ours :+1: 

It might be that what we need to do is be more active with **letting people know the gitter channel exists** and to **encourage more chat there**!

**Comment by enociz** [comment]
I don't like Gitter's experience but I'm fine with using it. I guess a Gitter [badge](http://GitHub.com/dwyl/repo-badges) should do.

**Comment by iteles** [comment]
I agree that the experience is certainly lacking unfortunately, but it's working better than slack for us _for now_. There should be a gitter badge visible when you open the start-here repo.

Do you see something different to the below?
<img width="384" alt="screen shot 2015-10-04 at 20 48 51" src="https://cloud.githubusercontent.com/assets/4185328/10269845/91bb6e14-6ad9-11e5-8df2-bc4aa6d47066.png">

**Comment by enociz** [comment]
Nope,

![Purrfect](http://images.techtimes.com/data/images/full/54926/cattambourine-gif.gif?w=600)
Purrfect!

**Comment by iteles** [comment]
Ha, yes, it's been there for a while now :wink: 

Thanks again for raising this!

**Comment by nelsonic** [comment]
Hi @enociz yeah, we prefer the *UX* of ***Slack***, 
but we :heart: the *people* [@GitterHQ](https://github.com/gitterHQ)
@mydigitalself @suprememoocow so we will continue to promote & use their work.

We *especially* :heart_eyes: how https://github.com/gitterHQ #OpenSource *most* of their *product* code.
By contrast, https://github.com/slackhq only make the *interfaces* to their product open... the distinction is *subtle* but important...

If you have *specific* issues with the Gitter UI/UX I'm *sure* @trevorah will appreciate the feedback. :wink:

**Comment by enociz** [comment]
Love your appreciation of #OpenSource 😀 keep it up

**Comment by edwardcodes** [comment]
@enociz (*we think*) #**OpenSource** is the #***Solution*** https://github.com/dwyl/intellectual-property :+1:

**Comment by nelsonic** [comment]
@markwilliamfirth apparently, I don't _have_ an account on slack for dwyl:
![image](https://cloud.githubusercontent.com/assets/194400/25751165/fc800710-31ab-11e7-9dc6-dcfdc5db9b81.png)

**Comment by newswim** [comment]
I'd love it if there were a DWYL slack channel, I tend to not use gitter for a few reasons (already having slack open, annoying notifications, somewhat awkward room system). +1 for slack!

**Comment by nelsonic** [comment]
@newswim while I _like_ the UI/UX of slack, 
I find that _other_ people (_who aren't getting work done_) 
use Slack  to distract the those who _are_ trying to get (focussed) work done... 😞  
![image](https://cloud.githubusercontent.com/assets/194400/25755346/8f71b3b2-31ba-11e7-8b15-7d88a2180ff8.png)
Read the comments on this: https://twitter.com/iamdevloper/status/857193510834982912

Slack is _meant_ to _boost_ people's "_**productivity**_" 
but it ends up being a _massive_ drain in practice. 
This is well documented in several academic studies; 
to get (_valuable_) work done we _must focus_, 
IM _obliterates_ any attempt at focus.

The argument proponents of IM use is 
that they can contact their colleagues and get answers much faster.
(_I totally get this!_)
What _most_ proponents of IM fail to acknowledge 
is that instantly-contacting your co-worker/peer interrupts their flow.
And it might take them **25 mins** to get back into their "flow".

Read: 
+ https://ux.useronboard.com/slack-i-m-breaking-up-with-you-54600ace03ea
+ https://medium.com/planet-arkency/3-easy-tricks-to-be-less-distracted-when-using-slack-a22b7791b9a6
+ https://www.groovehq.com/blog/stop-getting-distracted

Instead of thinking about how much work they ***could*** get done if they had **no interruptions**,
people concentrate on how "_useful_" IM was in an the _handful_ of cases where it helped.

> _Instant messaging is great for "**instant gratification**" 
but **terrible** for long-term success._
> It's _essential_ not to confuse "socialising" with peers via IM
with getting **high value** tasks done. 
Most people don't make the distinction.

If you want to be _highly productive_, follow Cal Newport's advice and use instant messaging as little as possible:
[![image](https://cloud.githubusercontent.com/assets/194400/22543046/9fe242a2-e926-11e6-86b2-21dbc91d5834.png)](https://youtu.be/zfoCyFvADtU)
https://youtu.be/zfoCyFvADtU

There is a _better_ way, we are going to _build_ it. 
A way of getting answers to questions in a timely fashion,
where content is well organised, structured and searchable.
A tool that _protects_ people's focus from interruption and
informs them of the highest priorities requiring their attention 
when they do take a break from their highest value tasks.
❤️  ✅

**Comment by newswim** [comment]
I love this approach and the idea!

There is a really interesting project which tries to make daily/weekly meetings more organized, (https://github.com/ParabolInc/action)
They've picked a great stack, and have an equity-based system for encouraging OSS contribution.

I agree that there are huge flaws with chat and "chat for business" software, many of which were inherited from the earlier days, and few products have come along which address these.

To add to your last statement, I would _really love to see_:

- A single view with relevant information (who has tried contacting me, what is that priority?)
- Users should be able to send messages with a priority indicator, only truly important things should be the cause of broken focus. (Of course this requires an understanding of "truly important")
- Automatic responses to common questions

Basically something intelligent that values the time of people using it, and helps to maximize productivity along the way.

Sounds like a great project for DWYL!!! 😍

**Comment by nelsonic** [comment]
"_**Chat** is great for occasional quick hashing things out, but when you leave it open all day in the corner of your screen, you're just asking for yourself **not** to be **working**. It's an **all-day-meeting** with **random participants** and **no agenda**_". ~ Jason Fried https://youtu.be/ft-q9s2uxIQ?t=14m43s

**Comment by iteles** [comment]
:arrow_up: That is a ***amazing*** way of putting it!

**Comment by markwilliamfirth** [comment]
@newswim there actually is a dwyl slack (http://dwylhq.slack.com/) it's just only used my myself and @iteles at the moment. We use it for rapid communication and when things aren't appropriate to share on GitHub. We have a number of private channels referring to different areas of the business that help organise and separate discussions. Each channel has a link to a google document related to the context of that channel. Personally I find the UX/UI of Slack much better than Gitter. It's certainly not perfect, but it's one of the better tools out there. It's likely that dwyl won't be building a chat product for a long time. We still have to build https://github.com/dwyl/time https://github.com/dwyl/tudo and https://github.com/dwyl/dwylbot first.

**Comment by iteles** [comment]
@newswim For reference @markwilliamfirth and I are using Slack as a trial and not using it for widespread dwyl communication - that is all done through gitter at the moment.

Personally, I _intensely dislike_ using Slack. My _biggest_ issues with it are:
- No 'read' notification so you never get the peace of mind that someone has looked at what you have said unless they take the time out to actively acknowledge it (I never use it for urgent comms but for example, if I mention that I'm going to be working from a different location the next day, I would like to know people have seen that message without having to tap them on the virtual shoulder to double check)
- What feels like dozens of clicks to switch between orgs and conversations, particularly on the phone
- It _CHEWS_ through battery on my phone, way more so than gitter
- The unread count on my phone often takes minutes to go away and is usually incorrect, wasting my time
- Does not load any content when the network is anything below perfect

**Comment by markwilliamfirth** [comment]
@iteles I get the impression from your comment above that you dislike the application and would like us to cease using it?

**Comment by newswim** [comment]
It's good to accumulate these levied issues, hopefully they roll into the product definitions for this _shiny new **BETTER**_ thing 😉 

I can echo the points @iteles makes, and think an OSS solution which addresses them would be a godsend. A productivity app that actually helps with productivity.

**Comment by iteles** [comment]
A note here also that I edited @markwilliamfirth's comment above to remove a reference to potential 'very long' timelines for the chat application - we haven't quite defined these yet and I didn't want us to be misleading, sorry @markwilliamfirth 😊

**Comment by nelsonic** [comment]
@newswim _funny_ you should mention "_product definitions_" ... 😉 
I was _just_ in the process of writing up a "User Story" 
for the _one_ thing I _miss_ from `Slack` when I'm using _other_ chat apps ...
see: https://github.com/dwyl/chat/issues/33

Slack has got _several_ things "_right_", Stewart Butterfield and his team have built a [_really good product_](https://www.goodreads.com/quotes/701885-good-is-the-enemy-of-great-and-that-is-one)!
But they have _also_ got _several_ things _horribly_ wrong and ... 
"_as a user_" I am unable to make improvements to "closed source" products 
even if I am technically competent and could _easily_ add the feature 
while following _all_ the necessary coding standards/workflow etc.
i.e. Closed Source software is _fundamentally_ flawed and _we_ are going to make it _obsolete_.
(_Yes, I know the irony of this statement considering the fact that the "**core**" of **GitHub** is very much **Closed**! But, I use GitHub to get my actual **work done** whereas "chat" is merely a distraction until I can sync it with my **work** and make people aware of the fact that I am in the middle of something **difficult** without having to **manually** update my "status" ... I/we can **easily** leave GitHub, but I don't think the alternative (GitLab/BitBucket/etc.) are "better", yet! 
The **reason** I **don't want** to be "**locked in**" to a **closed source** chat app/system is that I would **lose "control"** of my workflow!  if I start using `Slack` for work, people will start **expecting** me to be "**always on/available**" and , anyone who has read "**Deep Work**" knows this is a **terrible idea**..._)

> In my _experience_ the **people who _really value_ their time 
_don't_ keep _chat_ apps/windows open**
because chat (_notifications_) _destroy_ any attempt at concentration. 
It's a _widely_ known fact that "***multi-tasking doesn't work***".
Which means that people ***either*** keep a chat window open, 
***or*** they focus on their _important tasks_.

So we _also_ need the ability to disable _all_ notifications for someone 
when they are in "focussed mode". see: https://github.com/dwyl/chat/issues/35
(_i.e. when you are working on a task and should not be interrupted unless it's an emergency!_)

**Comment by markwilliamfirth** [comment]
@nelsonic not sure if you're aware of the do not disturb feature?

https://get.slack.help/hc/en-us/articles/214908388-Do-Not-Disturb-and-snooze-settings
https://get.slack.help/hc/en-us/articles/214888418-Set-default-Do-Not-Disturb-hours
https://api.slack.com/

**Comment by nelsonic** [comment]
@markwilliamfirth not sure if you _read_ the issue I opened: https://github.com/dwyl/chat/issues/35 (_and linked to in my previous comment_) ... Not only am I _aware_ of the `Slack` API but I have built an _integration_ using the API _and_ seen it's _limitations_!!

## _Incomplete_ Feature

Yes, `Slack` allows users to `set` "**Default Do Not Disturb Hours**" which is _good_. (_on the surface..._)
![image](https://cloud.githubusercontent.com/assets/194400/26238784/7adf228c-3c72-11e7-9fa0-18a5bb430422.png)

![image](https://cloud.githubusercontent.com/assets/194400/26242363/12b09458-3c80-11e7-8ddc-cbdd44d894e4.png)
> _Does it sound like a **good** thing that an "Admin" can "**override team members' own custom settings**"? Does that seem like a way to **protect** people's focus time or **dictate** when they **must** be available_?

## _Miscommunication_

![image](https://cloud.githubusercontent.com/assets/194400/26238761/612a0ece-3c72-11e7-9892-b73591d7ff1a.png)
`Slack` offering a _crude_ way to set a "Do Not Disturb" is only ***partially*** addressing the problem. 
And, as we all know, ***partially*** addressing does **not _solve_** the **problem**; _it often makes it **worse**_!
This leads to _miscommunication_ with/from your colleagues: 
+ "_But **you didn't set** your **Do Not Disturb**, so I **didn't know** you were **busy**_"... or 
+ "_Your `Slack` profile **always** says **Do Not Distrub**, are you **avoiding teamwork**_?!"

When _actually_ you have a _deadline_ to deliver something important that requires your _full attention_.
People shouldn't have to _manually_ set "Do Not Disturb" ... that's **_more_ work** (_for the user_)!!

## Clicks Cost _Real_ Money!

Let's run some _quick_ numbers for fun ... (_and so we can **close** this issue!_)

`Slack`'s token implementation of the "Do Not Disturb" or "Snooze Notification" feature 
_doesn't_ allow you to be "_flexible_" without additional _manual_ input/configuration.
While I would _love_ to have a **100% _Fixed_ Schedule** that _never_ deviates 
we can _all_ agree this is **_totally_ unrealistic** in **practice**. 

> `if` it takes an _average_ person **2 seconds** to ***manually update*** 
their "**Snooze Schedule**" **10 Times per day**<sup>1</sup> a day, it's **20 seconds**. 
_**20 seconds** **per person per day**, surely that's "**not-a-big-deal**" right..._?
Yes, it sounds "_insignificant_",  but ... 

+ _Multiply_: **2 seconds** x a **1,000 people**<sup>2</sup> (_i.e. **20k sec** or **5.5 hours** per day_)
+ _Multiply_: **5.5 hours** x $20/hour<sup>3</sup> (**$110/day**)
+ _Multiply_: **$110/day** x **260 working days** in a year that's ***$28,600/year***
and _suddenly_ this poorly-implemented feature is ***costing*** the company _real_ money!
does that _still sound_ like a _good_ use of time/money to you...?

### This _ONE Feature_ is Worth 1/3 `Slack`'s _ENTIRE Revenue_ for the _Year_!!!

> Slack has [1.25 Million Paid Users and $100M/year in Revenue](https://mattermark.com/slacks-growth-numbers)
> Sticking with our previous _reasonable assumption_ that it takes people ***20 seconds per day*** to set their "Snooze Notification" or "Do Not Disturb" settings (_trying to stay productive and avoid interruptions by communicating with their colleagues when they are/not available_)
_average_ people are paid **$20/hour** and **260 working days per year** ... 
this is **5,200 seconds / year** or ***87 minutes*** or **1.44 Hours**. 
+ Multiply: **1.44 hours** x **$20/hour** we get **$28.88 per person per year**
+ Multiply: **$28.88/person** x **1.25M**

Result: ***$36.1 Million / Year*** of ***wasted time*** on updating Snooze Settings.

## _Automatic_ & _Effortless_ [_Saves_ People `Time`](https://github.com/dwyl/start-here/blob/master/manifesto.md#your-time-should-never-be-wasted)

What I am _proposing_ is that we make the "Do Not Disturb" **Automatic** and ***Effortless*** based on the person's _current/active_ Task/Timer so that the communication tool helps people communicate _effectively_. _Not_ force them to perform more repetitive clicking. 

> _Yes_, We _could_ use the Slack API to do this, _however_, as I pointed out in the issue I created _yesterday_ (_`before` your API comment_), `if` _our_ product was even modestly successful we would _rapidly_ hit `Slack` "API Rate Limits" and then be _beholden_ to a Company with an Opaque/Fickle decision making process to grant us more API Requests...!

## Do You _Want_ To Build Features for a _Closed_ Platform...?

_Also_, `if` we build a _feature_ for the `Slack` Platform using their API we are doing **3 things**:

+ Helping to perpetuate `Slack`'s closed-source dominance
+ Limiting our own creativity by the limits (_both function and rate_) of their API
+ Giving them a ("_killer_") feature they can integrate into their core product and _eliminate_ us.

> Yes, We would "_get access to their 1.25M customer base_" as they point out in their API marketing... but those customers still _belong_ to `Slack`, they are merely giving us "access" to the users, which they can _easily_ `revoke` at _any_ point and release the _feature_ themselves.

## Stop Following the _Herd_ and `make` Something `New` / _Better_!

@dwyl's plan is _not_ to follow the "_herd_".

Focusing on how _good_ `Slack` is _guarantees_ that we won't realise how much _better_ it _could_ be! 
And even `if` we _do_ spot how much _better_ it could/should be, we are _powerless_ to improve it because it's ***CLOSED***. 

I don't _understand_ why we are _still_ _discussing_ this...
I thought we had made it _clear_.
We are _not_ going to use `Slack`. 
We are _going_ to make "instant messaging" (_interruption_) ***Obsolete***!
We cannot _build_ the future if we are _stuck_ using the _tools_ from the past!

How many people at `Apple` use `Nokia` Phones?
How many people working at `Amazon` buy their books from `Barnes and Noble`?
How many people working for `Nike` are wearing `Reebok` trainers?

I don't know how `else` to _communicate_ this...?
If I say: "_We are building a '**COMPETITOR**' to `Slack`_"!
People will _immediately_ proclaim that I'm "***insane***"! 
***Why*** would _anyone_ want to "_compete_" with `Slack`?!? 
_Surely_ they have "_solved_" the problem of "Chat"...?!?!?!?! _Right_....?!?
NO! The _issue_ is that `Slack` have _focussed_ on the ***wrong*** **problem**!!
They _should_ have solved the "**_Communication_ Problem**" not the _Instant Gratification_ One! 
***Communication*** is _way_ more than just having an Instant Messaging Client with an API.
Chat is only one _tiny_ part in ***Communication*** and it's the most _disruptive/distracting_ part!

I _agree_ that `Gitter` isn't great. 
Yes, I don't like `Gitter`'s notification system and "weird room" system. 
I _agree_ that we need a _better_ way to communicate @dwyl!!
But what we ***have*** to _agree_ on is that having _multiple_ forms of communication is ***Chaos***!
Email, Chat (_`Gitter`_), Google Docs, `GitHub`, Hangouts messages, SMS, WhatsApp, etc...
Too many _channels_ is a _recipe_ for _miss-communication_.
We _aren't_ going to _replace_ `Gitter` with `Slack`. 
What we _are_ going to do is _build_ a _simple_ ***Communication System***  
that _protects_ the time of the Most Valuable People in the team from _constant interruption_. 

Let's ***focus*** on _building_ a ***great*** **product** not perpetuating someone `else`'s business!

@markwilliamfirth _please_ focus your energy on helping us to `make` _our_ product: https://github.com/dwyl/chat/issues/37
(_or become a sales-person for someone `else` if you don't want to help `make` anything `new`...!_)

**Comment by markwilliamfirth** [comment]
@nelsonic the point is that before we're able to build the communication system we need an effective stopgap that we can use at dwyl.

The Slack do not disturb feature aligns with the premise of Deep Work and the Focus Hub offering - it allows for rhythmic periods of focus that are set and agreed with everyone. This means everyone knows when it is ok to talk and time to focus. The setting only needs to be set once from the admin console to be effective across the whole organisation.

Users dynamically altering the focus / communication state according to tasks is still quite a way off, whereas this is an effective stopgap that aligns with the deep work ethos, our offering and short-term vision for productivity.

**Comment by nelsonic** [comment]
@markwilliamfirth, you are _forcing_ me to make a decision I don't _want_ to make. 😞 
And I'm **NOT** talking about `Slack`.

**Comment by nelsonic** [comment]
https://abe-winter.github.io/plea%27s/help/2018/02/11/slack.html
![image](https://user-images.githubusercontent.com/194400/36097710-fef56068-0ff3-11e8-8213-11f7332ce8bc.png)

**Comment by newswim** [comment]
So true, even though I am basically forced to use Slack every day. The devil you know...

**Comment by nelsonic** [comment]
"Unslacking": https://beberlei.de/2018/10/28/unslacking_tideways_company.html
comments: https://news.ycombinator.com/item?id=18320968

**Comment by nelsonic** [comment]
[![image](https://user-images.githubusercontent.com/194400/51832778-8413e280-22ed-11e9-9c00-18d75352bce9.png)](https://twitter.com/iamdevloper/status/926458505355235328)

**Comment by nelsonic** [comment]
Slack Is Not Where 'Deep Work' Happens: [blog.nuclino.com/slack-is-not-where-deep-work-happens](https://blog.nuclino.com/slack-is-not-where-deep-work-happens)
Discussion thread: https://news.ycombinator.com/item?id=19643298

**Comment by nelsonic** [comment]
Read the top comment in this thread: https://news.ycombinator.com/item?id=23297113
![image](https://user-images.githubusercontent.com/194400/82784321-331bec00-9e58-11ea-8b2b-1a5ee9738248.png)

> "_Slack has totally become a drag on productivity and a net negative. Channel and message overload is arguably not much better than the email system it is meant to replace._"

>"_My prediction is that Slack will be replaced quite quickly by the first tool that really nails remote messaging and coordination. Slack is not it._"

**Comment by nelsonic** [comment]
This is what's wrong with having multiple _closed_ channels of communication: https://github.com/outmoded/discuss/issues/643#issuecomment-372099548
I'm trying to understand _why_ I'm seeing an error in the **`shot`** library for https://github.com/dwyl/labels/issues/92

How _helpful_ is this response (_and immediate closing_) of the issue?
![image](https://user-images.githubusercontent.com/194400/89103327-e0871f00-d408-11ea-8b84-b2d76f581299.png)

This issue was linked to from: https://github.com/outmoded/hapijs.com/issues/571
![image](https://user-images.githubusercontent.com/194400/89103361-3360d680-d409-11ea-956d-d78f35cb3809.png)

This is _super_ beginner-friendly. 🤦 

"[Outmoded](https://github.com/outmoded)" indeed. 👍
